packages/rfw/README.md

```diff
@@ -208,7 +328,542 @@ class _ExampleState extends State<Example> {
 }
 ```
 
-## Fetching remote widget libraries remotely
+Widgets in local widget libraries are represented by closures that are
+invoked by the runtime whenever a local widget is referenced.
+
+The closure uses the
+[LocalWidgetBuilder](https://pub.dev/documentation/rfw/latest/rfw/LocalWidgetBuilder.html)
+signature. Like any builder in Flutter, it takes a
+[`BuildContext`](https://api.flutter.dev/flutter/widgets/BuildContext-class.html),
+which can be used to look up inherited widgets. For example, when
+using the `createCoreWidgets` widgets,
+[`AnimationDefaults.curveOf`](https://pub.dev/documentation/rfw/latest/rfw/AnimationDefaults/curveOf.html)
+and
+[`AnimationDefaults.durationOf`](https://pub.dev/documentation/rfw/latest/rfw/AnimationDefaults/durationOf.html)
+can be used with the given `BuildContext` to get default curve and
+duration settings for animations.
+([`AnimationDefaults`](https://pub.dev/documentation/rfw/latest/rfw/AnimationDefaults-class.html)
+is a widget that the the `createCoreWidgets` local widget library
+introduces to allow such defaults to be set globally in a remote
+widget library. It defaults to 200ms
```

Comment:
Having reread this paragraph a couple of times and then looked at the code, I think this is a bit like `GestureDetector` in that it's confusing (for the new-to-rfw reader) because of a pun between a `Widget` widget and a local widget.

* There's an `AnimationDefaults` which is a Flutter widget provided by this package. That's the thing that `AnimationDefaults.curveOf` and `AnimationDefaults.durationOf`, mentioned above, are attached to.

* There's an `AnimationDefaults` which is a local widget introduced by `createCoreWidgets`. Naturally it's a wrapper for the other `AnimationDefaults`.

There isn't strictly any requirement to be using `createCoreWidgets` in order to use `AnimationDefaults.curveOf` and `AnimationDefaults.durationOf`. A LocalWidgetBuilder can always just use them, and in the worst case they'll always get the defaults (because there will be no enclosing `AnimationDefaults` in the Flutter tree). And even if someone prefers to build their local widgets library from scratch and skip `createCoreWidgets`, if they think `AnimationDefaults` is a good idea then it's just a few lines of code for them to wrap it in their own local widgets library so they can use it in remote widgets and those lookups will actually be useful.

For that matter, is it really expected that a local widget definition would use `AnimationDefaults.curveOf` or `AnimationDefaults.durationOf` directly? IIUC from the API docs, the idea is to instead use `ArgumentDecoders.curve` and `ArgumentDecoders.duration`.

Potentially the cleanest solution might be to find a different example for the use of `context`, and leave the use of `AnimationDefaults` to be explained elsewhere. From a quick grep of the code, the other example that shows up within the package would be `Directionality.of(context)`.

Reply:
yeah, fair. fixed.